### DISCUSSION TITLE
Can't build

### DISCUSSION THREAD

**Discussion by lazyclaw-labs** [CeresDB/ceresdb#182] [Q&A] [answered]
error[E0432]: unresolved import `upstream`
 --> grpcio\src\lib.rs:3:9
  |
3 | pub use upstream::*;
  |         ^^^^^^^^ use of undeclared crate or module `upstream`

**Comment by waynexia** [comment, marked as answer]
Which OS are you using? We currently don't support compiling on Windows (supported platform: https://github.com/CeresDB/ceresdb/blob/main/README.md#platform-support). I guess this conditional flag causes your problem:
https://github.com/CeresDB/ceresdb/blob/e9f7fa6c34c256eabe474f58e649768ff37e5b69/grpcio/Cargo.toml#L12-L16

**Comment by waynexia** [comment]
This dep (`grpcio`) says it can be built on windows (https://github.com/tikv/grpc-rs#prerequisites) but we haven't tried it... Sorry for the inconvenience.

**Comment by lazyclaw-labs** [comment]
yes, I am building on windows. Maybe that's the problem.

**Reply by archerny** [reply]
The dependency (grpcio) was removed recently(replaced by tonic), but the Windows platform is still not tested.
